src/data/files.rs

```diff
@@ -1,19 +1,41 @@
 use crate::prelude::*;
 use nu_errors::ShellError;
 use nu_protocol::{TaggedDictBuilder, UntaggedValue, Value};
+use std::path::PathBuf;
 
 pub(crate) fn dir_entry_dict(
-    filename: &std::path::Path,
-    metadata: &std::fs::Metadata,
+    path: &PathBuf,
     tag: impl Into<Tag>,
     full: bool,
+    short_name: bool,
 ) -> Result<Value, ShellError> {
-    let mut dict = TaggedDictBuilder::new(tag);
-    dict.insert_untagged("name", UntaggedValue::string(filename.to_string_lossy()));
+    let tag = tag.into();
+    let metadata = match std::fs::metadata(path) {
+        Ok(m) => Ok(m),
+        Err(e) => Err(ShellError::from(e)),
+    }?;
+    let file_type = metadata.file_type();
 
-    if metadata.is_dir() {
+    let mut dict = TaggedDictBuilder::new(&tag);
+
+    let name = if short_name {
+        match path.file_name() {
+            Some(n) => Ok(n.to_str().expect("This path was not properly encoded.")),
```

Comment:
`to_str()` returns `Option<OsStr>`. I think if the `?` is used, it would make that part of the if return `Ok(None)`. I could probably just put in an actual error if you want.

Reply:
I think an actual error would be great 👍